backend/module/eCampCore/src/EntityService/AbstractEntityService.php

```diff
@@ -118,18 +119,41 @@ public function fetchAll($params = []) {
     }
 
     /**
+     * Calls createEntity + is responsible for permission check and persistance.
+     *
      * @param mixed $data
      *
      * @throws NoAccessException
      * @throws ORMException
      *
-     * @return ApiProblem|BaseEntity
+     * @return BaseEntity
      */
     public function create($data) {
         $this->assertAllowed($this->entityClassname, __FUNCTION__);
-        $entity = $this->createEntity($this->entityClassname);
-        $this->getHydrator()->hydrate((array) $data, $entity);
+
+        $entity = $this->createEntity($data);
+
+        $this->assertAllowed($entity, __FUNCTION__);
+
         $this->serviceUtils->emPersist($entity);
+        $this->serviceUtils->emFlush();
```

Comment:
Diese Zeile macht zumindest den OAuth Login kaputt. Dort wird gleichzeitig eine `User` entity und eine `UserIdentity` entity erstellt, die sich gegenseitig kennen müssen. Durch direktes Flush jeder einzelnen entity beim createn entsteht eine Database Constraint Violation weil `userId` auf der `UserIdentity` nicht `null` sein darf.

Reply:
Ja, Flush ist heikel... kann nicht an jeder stelle aufgerufen werden.
Für die ActivityContents scheint es dort sinn zu machen... für OAuth nicht.

Die Problematik ist, solang eine neue Enitität nicht Flushed ist, kann diese per Select nicht von der DB gelesen werden.

Daher glaube ich, ein Flush müsste NACH den Services aber VOR dem Rendern platziert werden.

Reply:
Merci fürs Entdecken.

Das ist genau das Problem: Solange nicht geflushed ist kannst du nicht per select lesen, d.h. alle queries finden die neue Entität nicht. Vor dem Rendern reicht meiner Meinung nach nicht, weil evt. ein anderer Service davon abhängig ist, dass die Daten in der DB stehen (siehe Beispiel ActivityContent).

Das Problem liegt darin, dass wir halbfertige Objekte persisten. Das sollten wir nicht tun (bad practice). Wenn wir uns daran halten, dass wir keine halben Objekte persistieren, dann sind wir auch weniger fragil gegenüber flush-Fehler und Query-Fehler.

UserIdentity ist nun gefixt: Die Identity wird gleich im ersten Schritt komplett angelegt inkl. Relation zum User. Gut möglich, dass in dern anderen Services noch ein Fehler drin steckt. Bin da noch nicht alle durch.